plugins/text-freetype2/text-functionality.c

```diff
@@ -572,14 +588,17 @@ uint32_t get_ft2_text_width(wchar_t *text, struct ft2_source *srcdata)
 
 		load_glyph(srcdata, glyph_index, get_render_mode(srcdata));
 
-		if (text[i] == L'\n')
+		if (text[i] == L'\n') {
+			if (!all_lines) {
+				return w;
+			}
 			w = 0;
-		else {
+		} else {
 			w += slot->advance.x >> 6;
 			if (w > max_w)
 				max_w = w;
 		}
 	}
 
-	return max_w;
+	return all_lines ? max_w : w;
```

Comment:
Is this necessary to switch by `all_lines`? If `all_lines` is true and there are multiple lines, I don't think the control reaches here. So, I think it's always ok to return `max_w`.